cli/command/image/history_test.go

```diff
@@ -0,0 +1,94 @@
+package image
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/docker/docker/cli/internal/test"
+	"github.com/docker/docker/pkg/testutil/assert"
+	"github.com/docker/docker/pkg/testutil/golden"
+	"io/ioutil"
+	"testing"
+	"github.com/docker/docker/api/types/image"
+	"github.com/pkg/errors"
+	"regexp"
+)
+
+func TestNewHistoryCommandErrors(t *testing.T) {
+	testCases := []struct {
+		name             string
+		args             []string
+		imageHistoryFunc func(img string) ([]image.HistoryResponseItem, error)
+	}{
+		{
+			name: "wrong-args",
+			args: []string{},
+		},
+		{
+			name: "client-error",
+			args: []string{"image:tag"},
+			imageHistoryFunc: func(img string) ([]image.HistoryResponseItem, error) {
+				return []image.HistoryResponseItem{{}}, errors.Errorf("somethign went wrong")
+			},
+		},
+	}
+	for _, tc := range testCases {
+		buf := new(bytes.Buffer)
+		cmd := NewHistoryCommand(test.NewFakeCli(&fakeClient{imageHistoryFunc: tc.imageHistoryFunc}, buf))
+		cmd.SetOutput(ioutil.Discard)
+		cmd.SetArgs(tc.args)
+		err := cmd.Execute()
+		assert.NotNil(t, err)
+		actual := err.Error()
+		expected := string(golden.Get(t, []byte(actual), fmt.Sprintf("history-command-errors.%s.golden", tc.name))[:])
```

Comment:
Ok, I just feel that I need to generalise this as much as possible. But I didn't know assert.Error does a contains , I'll change it.